### PR TITLE
Adds a new trait - Bro Skater

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Adds a new trait that : 
Gives you a pro skateboard and cosmetic sunglasses when spawning
Reduces the stamina drain and damage of using/crashing from skateboards and wheelchairs by 70%

## Why It's Good For The Game
It was requested on the forum.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/8b7471c3-3fc7-4925-be18-b0cf66ff5819

</details>

## Changelog
:cl:
add: Added a new 2 point trait - Bro Skater. Spawns you in with cosmetic sunglasses and a pro skateboard, reduces stamina damage from riding/doing tricks/crashing with the skateboard and similar vehicles. Also makes you automatically flip any cap you put on.
/:cl:

### CODE REVIEW

File: code/modules/vehicles/scooter.dm [hunk 1]
```diff
@@ -100,18 +100,22 @@
...
 		playsound(src, 'sound/effects/bang.ogg', 40, TRUE)
 		if(!iscarbon(H) || H.getStaminaLoss() >= 100 || grinding || world.time < next_crash)
 			var/atom/throw_target = get_edge_target_turf(H, pick(GLOB.cardinals))
 			unbuckle_mob(H)
 			H.throw_at(throw_target, 3, 2)
 			var/head_slot = H.get_item_by_slot(ITEM_SLOT_HEAD)
 			if(!head_slot || !(istype(head_slot,/obj/item/clothing/head/helmet) || istype(head_slot,/obj/item/clothing/head/hardhat)))
-				H.adjustOrganLoss(ORGAN_SLOT_BRAIN, 5)
-				H.updatehealth()
+				if(prob(multiplier * 100)) //pro skaters get a 70% chance to not get the brain damage
```
Comment (itsmeow) [inline code/modules/vehicles/scooter.dm:114]: TIL you get brain damage from crashing

File: code/datums/traits/positive_quirk.dm [hunk 2]
```diff
@@ -251,3 +251,16 @@
...
+	name = "Skater Bro"
+	desc = "You’re a little too into old-earth skater culture! You're much more used to riding and falling off skateboards, needing less stamina to do kickflips and taking less damage upon bumping into something."
+	value = 2
+	mob_trait = TRAIT_PROSKATER
+	gain_text = "<span class='notice'>You feel like hitting a sick grind!</span>"
+	lose_text = "<span class='danger'>You no longer feel like you're in touch with the youth.</span>"
+
+/datum/quirk/proskater/on_spawn()
+	var/mob/living/carbon/human/H = quirk_target
+	H.equip_to_slot_or_del(new /obj/item/clothing/glasses/sunglasses(H), ITEM_SLOT_BACKPACK) //Totally radicool
```
Comment (Rukofamicom) [inline code/datums/traits/positive_quirk.dm]:
```suggestion
```

Sunglasses aren't necessary or entirely flavorful here imo

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
Overall looks good but the sunglasses (already available in coin shop iirc) are not necessary to the trait or flavor of the trait.

I'm mixed on whether this will be a truly beneficial addition too - it's always been pretty odd in a bad way to get someone skateboarding around the station.

File: code/datums/traits/positive_quirk.dm (hunk 2 above)
Comment (EvilDragonfiend) [reply code/datums/traits/positive_quirk.dm]: I agree with Ruko. You are fine to give subtype glasses that look funny instead.

File: code/modules/vehicles/motorized_wheelchair.dm [hunk 3]
```diff
@@ -156,8 +156,11 @@
 		var/atom/throw_target = get_edge_target_turf(H, pick(GLOB.cardinals))
 		unbuckle_mob(H)
 		H.throw_at(throw_target, 2, 3)
-		H.Knockdown(100)
-		H.adjustStaminaLoss(40)
+		var/multiplier = 1
+		if(HAS_TRAIT(H, TRAIT_PROSKATER))
+			multiplier = 0.3 //70% reduction
```
Comment (EvilDragonfiend) [inline code/modules/vehicles/motorized_wheelchair.dm]: 70% reduction seems too much for a 'motorized' wheelchair.

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
image

since you're here, would you fix this shit too?

**Comment by CydiaLamiales** [comment]
> Overall looks good but the sunglasses (already available in coin shop iirc) are not necessary to the trait or flavor of the trait.
> 
> I'm mixed on whether this will be a truly beneficial addition too - it's always been pretty odd in a bad way to get someone skateboarding around the station.

These sunglasses don't give flash protection, they're purely cosmetic.

**Review by EvilDragonfiend — APPROVED** [review]
I am okay with this. The most of code only does adding a flavourful skate quirk

**Comment by Rukofamicom** [comment]
As I clarified in discord: 

I am aware they do not have flash protection. Unrelated and purely cosmetic items do not belong as a part of any given trait, they belong in the coin shop.

**Comment by CydiaLamiales** [comment]
I'd argue that it's related, as it does fit the "Rad staker dude" style, but I'll remove it.

File: code/datums/traits/positive_quirk.dm [hunk 4]
```diff
@@ -251,3 +251,16 @@
...
+	name = "Skater Bro"
+	desc = "You’re a little too into old-earth skater culture! You're much more used to riding and falling off skateboards, needing less stamina to do kickflips and taking less damage upon bumping into something."
+	value = 2
+	mob_trait = TRAIT_PROSKATER
+	gain_text = "<span class='notice'>You feel like hitting a sick grind!</span>"
+	lose_text = "<span class='danger'>You no longer feel like you're in touch with the youth.</span>"
+
+/datum/quirk/proskater/on_spawn()
+	var/mob/living/carbon/human/H = quirk_target
+	H.equip_to_slot_or_del(new /obj/item/clothing/glasses/sunglasses(H), ITEM_SLOT_BACKPACK) //Cosmetic sunglasses that don't give flash protection
```
Comment (CydiaLamiales) [inline code/datums/traits/positive_quirk.dm]:
```suggestion
```

**Review by Rukofamicom — APPROVED** [review]
My only reservation now is whether or not we want to encourage players to be a skater bro character or not.

Also take my code review with a grain of salt given it is phone reviewed

**Comment by EvilDragonfiend** [comment]
Since this is a minor addition, I will merge this in advance. Please do not forget to add the feature to the wiki.

**Pull request merged**